systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java

```diff
@@ -69,6 +71,113 @@ void testProducerConsumer() {
         ClientUtils.waitForClientSuccess(OAUTH_CONSUMER_NAME, NAMESPACE, MESSAGE_COUNT);
     }
 
+    @Test
+    void testProducerConsumerAudienceTokenChecks() {
+        LOGGER.info("Setting producer and consumer properties");
+        KafkaOauthExampleClients oauthAudienceInternalClientJob = new KafkaOauthExampleClients.Builder()
+                .withProducerName(OAUTH_PRODUCER_NAME)
+                .withConsumerName(OAUTH_CONSUMER_NAME)
+                .withBootstrapAddress(KafkaResources.bootstrapServiceName(oauthClusterName) + ":" + audienceListenerPort)
+                .withTopicName(TOPIC_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withOAuthClientId(OAUTH_CLIENT_NAME)
+                .withOAuthClientSecret(OAUTH_CLIENT_SECRET)
+                .withOAuthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                .build();
```

Comment:
I guess that `oauthInternalClientJob` is almost the same -> only bootstrap address is different.
So, maybe you can just change the `oauthInternalClientJob`'s bootstrap address and then change it back. Or do something like:
```
KafkaOauthExampleClients oauthAudienceInternalClientJob = oauthInternalClientJob.toBuilder()..withBootstrapAddress(KafkaResources.bootstrapServiceName(oauthClusterName) + ":" + audienceListenerPort)
```
or something like that (I would prefer the first way).

Reply:
working on it. Thanks for idea.